colcon_core/shell/sh.py

```diff
@@ -26,6 +26,8 @@ class ShShell(ShellExtensionPoint):
     FORMAT_STR_USE_ENV_VAR = '${name}'
     FORMAT_STR_INVOKE_SCRIPT = 'COLCON_CURRENT_PREFIX="{prefix}" ' \
         '_colcon_prefix_sh_source_script "{script_path}"'
+    FORMAT_STR_REMOVE_TRAILING_SEPARATOR = 'if test "$(echo -n ${name} | ' \
+        'tail -c 1)" = ":" ; then; export {name}=${{{name}%?}} ; fi'
```

Comment:
I couldn't come up with a nicer expression... I think this is POSIX compliant though.